pkg/build/controller/config_controller.go

```diff
@@ -5,20 +5,59 @@ import (
 
 	"github.com/golang/glog"
 	kapi "k8s.io/kubernetes/pkg/api"
-	kerrors "k8s.io/kubernetes/pkg/api/errors"
-	"k8s.io/kubernetes/pkg/util"
+	kapierrors "k8s.io/kubernetes/pkg/api/errors"
+	kutil "k8s.io/kubernetes/pkg/util"
+	kerrors "k8s.io/kubernetes/pkg/util/errors"
 
 	buildapi "github.com/openshift/origin/pkg/build/api"
 	buildclient "github.com/openshift/origin/pkg/build/client"
+	"github.com/openshift/origin/pkg/build/util"
 )
 
 type BuildConfigController struct {
 	BuildConfigInstantiator buildclient.BuildConfigInstantiator
+	BuildConfigUpdater      buildclient.BuildConfigUpdater
+	BuildConfigDeleter      buildclient.BuildConfigDeleter
+	BuildDeleter            buildclient.BuildDeleter
+	BuildLister             buildclient.BuildLister
 }
 
 func (c *BuildConfigController) HandleBuildConfig(bc *buildapi.BuildConfig) error {
 	glog.V(4).Infof("Handling BuildConfig %s/%s", bc.Namespace, bc.Name)
 
+	if !bc.DeletionTimestamp.IsZero() {
+		if bc.Status.CanDelete {
```

Comment:
ok that makes sense.  but i think you can get rid of this field entirely.  all you need is the timestamp to indicate that the deletion has been requested.  every time handle build gets run, if the timestamp is set, you attempt to delete all the child builds.  and once all the child builds are deleted, you attempt to delete the buildconfig.

setting CanDelete is really just saving you the effort of checking if there are any child builds in the case where you already deleted all the child builds, but then failed to delete the buildconfig.